File: ems-backend/src/main/java/com/dabananda/ems_backend/service/DepartmentService.java
package com.dabananda.ems_backend.service;

import com.dabananda.ems_backend.dto.DepartmentDto;

import java.util.List;

public interface DepartmentService {
    DepartmentDto createDepartment(DepartmentDto departmentDto);
    DepartmentDto getDepartmentById(Long departmentId);
    List<DepartmentDto> getAllDepartments();
    DepartmentDto updateDepartment(Long departmentId, DepartmentDto updateDepartment);
    void deleteDepartment(Long departmentId);
}

File: ems-frontend/src/services/DepartmentService.js
import axios from "axios";

const DEPARTMENT_REST_API_BASE_URL = 'http://localhost:8080/api/departments';

export const getAllDepartments = () => axios.get(DEPARTMENT_REST_API_BASE_URL);

export const createDepartment = (department) => axios.post(DEPARTMENT_REST_API_BASE_URL, department);

export const updateDepartment = (departmentId) => axios.post(DEPARTMENT_REST_API_BASE_URL + '/' + departmentId);

export const getDepartmentById = (departmentId) => axios.get(DEPARTMENT_REST_API_BASE_URL + '/' + departmentId);